crates/sui-adapter-transactional-tests/tests/mvcc/find_all_uids_dof_enum.move

```diff
@@ -0,0 +1,207 @@
+    }
+
+    fun set_wrapped(w: &mut Wrapped, value: u64) {
+        set_(&mut w.id, value);
+        set_(w.other.get_wrapped_uid(), value);
+
+    }
+
+    fun set_(id: &mut UID, value: u64) {
+        ofield::borrow_mut<u64, Value>(id, KEY).value = value;
+    }
+
+    //////////////////////////////////////////////////////////////
+    // remove
+
+    public fun remove(s: &mut S) {
+        remove_(&mut s.id);
+        remove_(s.other.get_wrapped_uid());
+        remove_wrapped(&mut s.wrapped);
+        remove_wrapped(vector::borrow_mut(&mut s.many, 0));
+        remove_wrapped(vector::borrow_mut(&mut s.many, 1));
+    }
+
+    fun remove_wrapped(w: &mut Wrapped) {
+        remove_(&mut w.id);
+        remove_(w.other.get_wrapped_uid());
+    }
+
+    fun remove_(id: &mut UID) {
+        let Value { id, .. } = ofield::remove(id, KEY);
+        object::delete(id);
+    }
+
+    //////////////////////////////////////////////////////////////
+    // check
+
+    public fun check(s: &S, expected: Option<u64>) {
+        check_(&s.id, expected);
+        check_(s.other.get_wrapped_uid_ref(), expected);
+        check_wrapped(&s.wrapped, expected);
+        check_wrapped(vector::borrow(&s.many, 0), expected);
+        check_wrapped(vector::borrow(&s.many, 1), expected);
+    }
+
+    fun check_wrapped(w: &Wrapped, expected: Option<u64>) {
+        check_(&w.id, expected);
+        check_(w.other.get_wrapped_uid_ref(), expected);
+    }
+
+    fun check_(id: &UID, expected: Option<u64>) {
+        if (option::is_some(&expected)) {
+            let Value { value, .. } = ofield::borrow(id, KEY);
+            assert!(value == option::borrow(&expected), 0);
+        } else {
+            assert!(!ofield::exists_with_type<u64, Value>(id, KEY), 0);
+        }
+    }
+}
+
+//# programmable --sender A --inputs @A
+//> 0: test::m::new();
+//> TransferObjects([Result(0)], Input(0))
+
+//# view-object 2,8
+
+//# programmable --sender A --inputs object(2,8) 112
+//> test::m::set(Input(0), Input(1))
+
+//# view-object 2,8
+
+//# programmable --sender A --inputs object(2,8) 112
+//> test::m::remove(Input(0))
+
+//# view-object 2,8
+
+// dev-inspect with 'check' and correct values
+
+//# programmable --sender A --inputs object(2,8)@2 vector[0] --dev-inspect
+//> test::m::check(Input(0), Input(1))
+
+//# programmable --sender A --inputs object(2,8)@3 vector[112] --dev-inspect
+//> test::m::check(Input(0), Input(1))
+
+//# programmable --sender A --inputs object(2,8)@4 vector[] --dev-inspect
+//> test::m::check(Input(0), Input(1))
+
+
+// dev-inspect with 'check' and _incorrect_ values
+
+//# programmable --sender A --inputs object(2,8)@3 vector[0] --dev-inspect
+//> test::m::check(Input(0), Input(1))
+
+//# programmable --sender A --inputs object(2,8)@4 vector[112] --dev-inspect
+//> test::m::check(Input(0), Input(1))
+
+//# programmable --sender A --inputs object(2,8)@2 vector[] --dev-inspect
+//> test::m::check(Input(0), Input(1))
```

Comment:
Small nit: for future maintenance, can you explain which of these are expected to fail/pass in a comment (and why)?